src/core/rewriter.sk

```diff
@@ -315,6 +318,12 @@ class GLSLX.Rewriter {
       }
 
       case .IF {
+        if node.ifTrue == null {
```

Comment:
Hmm. It occurs to me now that a better fix might be to find out where we're removing the ifTrue from the .IF node, instead of handling the `node.ifTrue == null` case here.

Pros of that approach: maybe you consider an .IF node with no children to be an invalid AST. It's easier to reason through the code if every transformer (e.g. the code that compacts, the code that trims, etc.) each knows how to consume a valid AST and output a valid AST.

Downside of that approach: whenever you remove a node, you have to consider every possible case for what it might be and make sure that the resulting AST is valid.

Do you have thoughts on this?